### PR TITLE
fix: add peru_compras_bulk, restore peru_compras

### LINKED ISSUE #1120: peru_compras: update spider to use new endpoint
Hopefully it's temporary, as we need their data :)

Sentry Issue: [REGISTRY-KINGFISHER-COLLECT-1](https://open-contracting-partnership.sentry.io/issues/2823958328/?referrer=github_integration)

```
Gave up retrying <GET https://www.catalogos.perucompras.gob.pe/ConsultaOrdenesPub/DescargaJsonOCDS?pAcuerdo=151&pFechaIni=2017-01-01&pFechaFin=2024-11-01> (failed 3 times): 500 Internal Server Error
```

### PR DESCRIPTION
**Pull request by yolile**
closes #1120

### CODE REVIEW

File: kingfisher_scrapy/spiders/peru_compras.py [hunk 1]
```diff
@@ -1,49 +1,46 @@
 import scrapy
 
-from kingfisher_scrapy.base_spiders import SimpleSpider
-from kingfisher_scrapy.util import components, handle_http_error
+from kingfisher_scrapy.spiders.peru_compras_base import PeruComprasBase
+from kingfisher_scrapy.util import handle_http_error, parameters
 
 
-class PeruCompras(SimpleSpider):
+class PeruCompras(PeruComprasBase):
```
Comment (jpmckinney) [inline kingfisher_scrapy/spiders/peru_compras.py:7]: Should this now be called peru_compras_api to disambiguate from _bulk?

File: kingfisher_scrapy/spiders/peru_compras_bulk.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    date_format = 'year-month'
+    default_from_date = '2021-08'
+
+    def start_requests(self):
+        yield scrapy.Request(
+            f'{self.url_prefix}getListaDescargaMasiva?Anio=&Mes=',
+            method='POST',
+            meta={'file_name': 'list.json'},
+            callback=self.parse_list
+        )
```
Comment (jpmckinney) [inline kingfisher_scrapy/spiders/peru_compras_bulk.py:33]: To make it easy to compare the two spiders:

```suggestion
        url = f'{self.url_prefix}getListaDescargaMasiva?Anio=&Mes='
        yield scrapy.Request(url, method='POST', meta={'file_name': 'list.json'}, callback=self.parse_list)
```

File: kingfisher_scrapy/spiders/peru_compras.py (hunk 1 above)
Comment (yolile) [reply kingfisher_scrapy/spiders/peru_compras.py:7]: Sure, and I will change the data registry entry  when we deploy this

File: kingfisher_scrapy/spiders/peru_compras_api.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import scrapy
+
+from kingfisher_scrapy.spiders.peru_compras_base import PeruComprasBase
+from kingfisher_scrapy.util import handle_http_error, parameters
+
+
+class PeruComprasApi(PeruComprasBase):
```
Comment (jpmckinney) [inline kingfisher_scrapy/spiders/peru_compras_api.py:7]: Capitalize API like others.

**Review by jpmckinney — COMMENTED** [review]
Okay to merge once comments resolved.

File: kingfisher_scrapy/spiders/peru_compras_api.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    """
+    Domain
+      Peru Compras (contracts within framework agreements)
```
[the remark this answers is not in the document]
Comment (yolile) [reply kingfisher_scrapy/spiders/peru_compras_api.py:10]: Yes, it is

**Pull request merged**